src/OpenLoco/Audio/ChannelManager.cpp

```diff
@@ -0,0 +1,175 @@
+#include "ChannelManager.h"
+#include "VehicleChannel.h"
+#include <algorithm>
+#include <iterator>
+
+namespace OpenLoco::Audio
+{
+    static Channel* makeNewChannel(OpenAL::SourceManager& sourceManager, ChannelId channelId)
```

Comment:
Best to use unique_ptr, this is currently leaking.

Reply:
yeah I will, still trying to figure out why vehicle sounds aren't playing atm, then will fix the memory issues